jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunSettingsEditorPanel.kt

```diff
@@ -0,0 +1,167 @@
+            if (selectedRuntime == lastSelectedRuntime) return@addActionListener
+            lastSelectedRuntime = selectedRuntime
+            handlerPanel.setRuntime(selectedRuntime)
+        }
+        updateComponents()
+    }
+
+    private fun updateComponents() {
+        val handler = handlerPanel.handler
+        handlerPanel.isEnabled = !useTemplate.isSelected
+        runtime.isEnabled = !useTemplate.isSelected
+        templateFile.isEnabled = useTemplate.isSelected
+        timeoutSlider.setEnabled(!useTemplate.isSelected)
+        memorySlider.setEnabled(!useTemplate.isSelected)
+        if (useTemplate.isSelected) {
+            handler.background = UIUtil.getComboBoxDisabledBackground()
+            handler.foreground = UIUtil.getComboBoxDisabledForeground()
+            if (functionModels.selectedItem is Function) {
+                val selected = functionModels.selectedItem as Function
+                handler.text = selected.handler()
+                val memorySize = selected.memorySize()
+                val timeout = selected.timeout()
+                if (memorySize != null) {
+                    memorySlider.value = memorySize
+                }
+                if (timeout != null) {
+                    timeoutSlider.value = timeout
+                }
+                val runtime = Runtime.fromValue(tryOrNull { selected.runtime() })
+                runtimeModel.selectedItem = runtime.validOrNull
+                function.isEnabled = true
+            }
+        } else {
+            handler.background = UIUtil.getTextFieldBackground()
+            handler.foreground = UIUtil.getTextFieldForeground()
+            function.setEnabled(false)
+        }
+    }
+
+    fun setTemplateFile(file: String?) {
+        if (file == null) {
+            templateFile.text = ""
+            updateFunctionModel(emptyList())
+        } else {
+            templateFile.text = file
+            val functions = findFunctionsFromTemplate(project, File(file))
+            updateFunctionModel(functions)
+        }
+    }
+
+    private fun updateFunctionModel(functions: List<Function>) {
+        functionModels.removeAllElements()
+        function.isEnabled = functions.isNotEmpty()
+        functionModels.addAll(functions)
+        if (functions.size == 1) {
+            functionModels.setSelectedItem(functions[0])
+        } else {
+            function.setSelectedIndex(-1)
+        }
+        updateComponents()
+    }
+
+    fun selectFunction(logicalFunctionName: String?) {
+        logicalFunctionName ?: return
+        val function = functionModels.find { it.logicalName == logicalFunctionName } ?: return
+        functionModels.selectedItem = function
+        updateComponents()
+    }
+
+    fun setRuntimes(runtimes: List<Runtime>?) {
+        runtimeModel.setAll(runtimes)
+    }
+
+    fun invalidateConfiguration() {
+        SwingUtilities.invokeLater { invalidator.isSelected = !invalidator.isSelected }
```

Comment:
why not runInEdit?

Reply:
This is what was here before, didn't change it. Changed to `runInEdt`